content/en/docs/reference/networking/virtual-ips.md

```diff
@@ -170,6 +170,24 @@ address.
 This same basic flow executes when traffic comes in through a node-port or
 through a load-balancer, though in those cases the client IP address does get altered.
 
+#### Performance improvements in the iptables proxy mode
+
+{{< feature-state for_k8s_version="v1.26" state="alpha" >}}
+
+In large clusters (with tens of thousands of Pods and Services), the
+iptables mode of kube-proxy may take a long time to update the rules
+in the kernel when Services change.
+
+The latest version of kube-proxy has some performance improvements in
+this code, but they are not enabled by default (and should probably
+not be enabled in production clusters yet). To try them out, enable
+the `MinimizeIPTablesRestore` [feature
+gate](/docs/reference/command-line-tools-reference/feature-gates/) for
+kube-proxy with `--feature-gates=MinimizeIPTablesRestore=true,…`. If
+you were previously overriding the `--iptables-min-sync-period`
+option, you should also try removing that override and letting it use
+the default value ("`1s`").
```

Comment:
I'm not keen on this. It'll date quickly (“the latest version” will be wrong as soon as v1.27 is out) and is framed as the _change_ more than documenting the _behavior_.

The heading should describe the optimization (eg: **Optimization for `iptables-restore`**) and let the reader infer that:
- we saw that as an improvement
- it applies to iptables mode

If the feature gate is really experimental, let's mention that right in the heading, but set a manual fragment identifier so that links to this still work for the beta. I can provide an example once we know what the heading text is going to be.

Reply:
> is framed as the _change_ more than documenting the _behavior_

Right, because KEP 3453 _is_ just a change, not a new feature. Once it goes to beta, there's really nothing to say about it any more.

(Yes, if we enable the feature gate by default in 1.28 and it breaks some clusters, then those users should file bugs and disable the feature gate. But if that happened, there is no reason they would end up finding this particular section of the documentation; the warning would need to be in the release notes, not here. And then once the feature goes to GA, there's definitely nothing to document here, since at that point, even if there are still bugs, they're just bugs in kube-proxy, not bugs in a particular feature.)

This is an unusual KEP, and it was not even originally supposed to have been a KEP. I had already implemented the change, and then Tim suggested that we put it through the KEP process, mostly because he wanted to make sure that we had thought about all of the things that the KEP template forces you to think about. (And also so we'd have a feature gate to make it opt-in/out-able.)

----

I guess rather than documenting KEP 3453, we could add an "Optimizing the iptables mode" section, which would document the current behavior of `--iptables-sync-period` and `--iptables-min-sync-period`, and then mention the new feature gate, and how that changes the recommendations around those options. But when the feature goes GA, the section would mostly become a stub and just say "if you used to be overriding these options, then you can probably stop".